styles/lib/carnival/features/_tables.scss

```diff
@@ -0,0 +1,367 @@
+@use 'lib/carnival/colors';
+@use 'lib/carnival/fonts';
+
+@mixin basic-table($border-after-table: $black) {
+    .os-table:not(.os-unstyled-container) {
+      margin-bottom: 0.7rem;
+      display: table;
+      margin-left: auto;
+      margin-right: auto;
+      > table {
+        border-collapse: collapse;
+        width: 100%;
+        margin-bottom: 0;
+        > thead > tr {
+          > th {
+            border-bottom: 0.1rem solid colors.$black;
+            border-top: 0.1rem solid colors.$black;
+            font-family: fonts.$mulish;
+            font-weight: 700;
+            padding: 0.7rem;
+          }
+          > th[data-valign=top] {
+            vertical-align: top;
+          }
+          > th[data-align=left] {
+            text-align: left;
+          }
+          > th[data-align=right] {
+            text-align: right;
+          }
+          > th[data-align=center] {
+            text-align: center;
+          }
+        }
+      }
+    }
+    .os-table:not(.os-unstyled-container) > table > tbody > tr.header-row > td {
```

Comment:
rather than restate this selector, you can nest it in the above styles starting with `tbody` at the same level as `thead`. same with the ones below it

Reply:
as far as i can tell this has not yet been resolved -- what i'm hoping for is a structure where the only place `.os-table:not(.os-unstyled-container)` appears is the first line of this mixin, & same for each subselector.

i guess another way of saying it is, i'm hoping to write these mixins such that subselectors don't repeat, if possible. that means that this:

```scss
.os-table:not(.os-unstyled-container) > table > tbody > tr.header-row > td {
  // specs 1
}

.os-table:not(.os-unstyled-container) > table > tbody > tr.header-row > td:not(:only-of-type) {
  // specs 2
}

.os-table:not(.os-unstyled-container) > table > tbody > tr {
  > td:not(:only-of-type) > strong {
    // specs 3
  }
  > td[rowspan="2"]:not(:only-of-type):not(:first-child) {
    // specs 4
  }
  > td:not(:only-of-type):not(:last-child) {
    // specs 5
  }
  > td {
    // specs 6
  }
}
```

becomes this:

```scss
.os-table:not(.os-unstyled-container) > table > tbody > tr {
  &.header-row > td {
    // specs 1
    &:not(:only-of-type) {
      // specs 2
    }
  }
  > td {
    // specs 6
    &:not(:only-of-type) > strong {
      // specs 3
    }
    &[rowspan="2"]:not(:only-of-type):not(:first-child) {
      // specs 4
    }
    &:not(:only-of-type):not(:last-child) {
      // specs 5
    }
  }
}

```